### ISSUE TITLE
Deprecate is_extension_type

### ISSUE THREAD

**Issue by TomAugspurger** [pandas-dev/pandas#19041]
In favor of is_extension_array_type. This is blocked till sparse, period, and datetimetz are all EAs.

---

This should probably be True

```python
In [3]: pd.api.types.is_extension_type(pd.interval_range(0, 10))
Out[3]: False
```

Likewise for Period?

```python
In [3]: pd.api.types.is_extension_type(pd.period_range(2017, periods=2))
Out[3]: False
```

Currently we hardcode checks for categorical, sparse, and datetime w/ tz. A general solution would be to have all our extension types inherit from some base class or satisfy the interface required for an extension type.

xref https://github.com/pandas-dev/pandas/issues/18767

**Comment by toobaz** [comment]
If i understand correctly, @TomAugspurger you are (in today's terms) just suggesting that ``is_extension_type`` should return ``True`` if ``is_extension_array_dtype`` does, right? (Notice this is already true for ``Categorical``).

Then I think the only two hardcoded types would be ``datetimetz`` and ``sparse``.

By the way: is there any plan to make those extension types too? The fact that
```
In [2]: s = pd.date_range(start='2010-08-08', end='2010-08-09', tz='utc')

In [3]: s._values is s
Out[3]: True
```
is quite annoying in parts of the code like this:
https://github.com/pandas-dev/pandas/blob/ffae158749c9b72720a7fe3295d323082a5fe420/pandas/core/base.py#L942

which assume that the ``_values`` of an ``Index`` are not an ``Index`` themselves.

**Comment by TomAugspurger** [comment]
I think eventually we'll just remove `is_extension_type`, in favor of `is_extension_array_dtype`.

For context, changing everything in one go was too difficult, so we've kept `is_extension_type` for backwards compat. Once everything is an EA, we could remove it (after deprecating).

> By the way: is there any plan to make those extension types too? The fact that

(just replied on the mailing list). I just opened a PR for sparse.
